compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1479,6 +1565,14 @@ public void visit(BLangAssignment assignNode) {
     public void visit(BLangTupleDestructure tupleDeStmt) {
         for (BLangExpression tupleVar : tupleDeStmt.varRef.expressions) {
             setTypeOfVarRefForBindingPattern(tupleVar);
+            if (tupleVar.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
```

Comment:
What if it is not simple var ref but in the member var refs has an invalid assignment?